File: app/src/main/java/com/ufkoku/ciphersharedpreferences/demo/dagger2/cipher_holder/CipherHolderModule.java
package com.ufkoku.ciphersharedpreferences.demo.dagger2.cipher_holder;

import com.ufkoku.cipher_sharedprefs.api.ICipherHolder;
import com.ufkoku.cipher_sharedprefs.cipher.AesCipher;

import java.util.Arrays;

import dagger.Module;
import dagger.Provides;

@Module
public class CipherHolderModule {

    private final byte[] key;

    public CipherHolderModule(byte[] key) {
        this.key = Arrays.copyOf(key, key.length);
    }

    @Provides
    public ICipherHolder provideCipher() {
        return new AesCipher(key);
    }

}

File: app/src/main/kotlin/com/ufkoku/ciphersharedpreferences/demo/dagger2/cipher_holder/CipherHolderModule.kt
package com.ufkoku.ciphersharedpreferences.demo.dagger2.cipher_holder

import com.ufkoku.cipher_sharedprefs.api.ICipherHolder
import com.ufkoku.cipher_sharedprefs.cipher.AesCipher

import java.util.Arrays

import dagger.Module
import dagger.Provides

@Module
class CipherHolderModule(val key: ByteArray) {

    @Provides
    fun provideCipher(): ICipherHolder {
        return AesCipher(key)
    }

}
